avm/client/src/avmHelpers.ts

```diff
@@ -83,15 +93,13 @@ export function deserializeAvmResult(result: any): InterpreterResult {
         let arguments_;
         let tetraplets;
         try {
-            const argumentsStr = decoder.decode(new Uint8Array(callRequest.arguments));
-            arguments_ = JSON.parse(argumentsStr);
+            arguments_ = argumentRepr.fromBinary(callRequest.arguments)
         } catch (e) {
             throw "Couldn't parse arguments: " + e + '. Original data is: ' + callRequest.arguments;
         }
 
         try {
-            const tetrapletsStr = decoder.decode(new Uint8Array(callRequest.tetraplets));
-            tetraplets = JSON.parse(tetrapletsStr);
+            tetraplets = tetrapletRepr.fromBinary(callRequest.tetraplets)
         } catch (e) {
             throw "Couldn't parse tetraplets: " + e + '. Original data is: ' + callRequest.tetraplets;
         }
```

Comment:
below you have `key as any`
Probably you need to rewrite it as Number(key).
`for ... in ` cycle gives only string key and judging by the output type you'd like to have numbers

Reply:
It worth changing it in a separate PR.